### PR TITLE
SCRUM-3698: update GH actions functions and mask password

### PR DESCRIPTION
**Pull request by abecerra**

### CODE REVIEW

File: .github/workflows/github-release-build-and-deploy.yml [hunk 1]
```diff
@@ -48,7 +50,7 @@ jobs:
           cat <(echo -e "${{ secrets.OSSRH_GPG_SECRET_KEY }}") | gpg --batch --import
           gpg --list-secret-keys --keyid-format LONG
       - name: Publish package
-        run: mvn --batch-mode deploy
+        run: mvn --batch-mode deploy -DskipITs=true
```
Comment (mluypaert) [inline .github/workflows/github-release-build-and-deploy.yml:53]: @abecerra According to maven docs, this should only be `-DskipITs` (so without the `=true`) https://maven.apache.org/surefire/maven-failsafe-plugin/examples/skipping-tests.html

File: .github/workflows/alpha-build-and-deploy.yml [hunk 2]
```diff
@@ -23,14 +23,16 @@ jobs:
...
         with:
           role-to-assume: ${{secrets.GH_ACTIONS_AWS_ROLE}}
           role-session-name: gh-actions-${{github.run_id}}.${{github.run_number}}.${{github.run_attempt}}-build-image
           aws-region: us-east-1
       - name: Amazon ECR login
         id: login-ecr
-        uses: aws-actions/amazon-ecr-login@v1
+        uses: aws-actions/amazon-ecr-login@v2
+        with:
+          mask-password: 'true'
```
Comment (mluypaert) [inline .github/workflows/alpha-build-and-deploy.yml:35]: @abecerra `mask-password` defaults to true in this aws-actions/amazon-ecr-login release, so you can remove the `with:` section here.

File: .github/workflows/github-release-build-and-deploy.yml [hunk 3]
```diff
@@ -9,16 +9,18 @@ jobs:
...
       with:
         role-to-assume: ${{secrets.GH_ACTIONS_AWS_ROLE}}
         role-session-name: gh-actions-${{github.run_id}}.${{github.run_number}}.${{github.run_attempt}}-build-image
         aws-region: us-east-1
     - name: Amazon ECR login
       id: login-ecr
-      uses: aws-actions/amazon-ecr-login@v1
+      uses: aws-actions/amazon-ecr-login@v2
+      with:
+        mask-password: 'true'
```
Comment (mluypaert) [inline .github/workflows/github-release-build-and-deploy.yml:23]: Same here, `with:` section can be removed as it simply restates the default value.

**Review by mluypaert — CHANGES_REQUESTED** [review]

File: .github/workflows/github-release-build-and-deploy.yml (hunk 1 above)
Comment (abecerra) [reply .github/workflows/github-release-build-and-deploy.yml:51]: According to quarkus guide (below the code block, in point 1), the skipITs must have the true value.
I guess we can test with the true value first and if it does not work, remove it.

https://quarkus.io/guides/maven-tooling#build-tool-maven

**Review by mluypaert — APPROVED** [review]

**Pull request merged**